apps/studio/components/interfaces/RoleImpersonationSelector/RoleImpersonationSelector.tsx

```diff
@@ -0,0 +1,133 @@
+import { useState } from 'react'
+
+import { PostgrestRole } from 'lib/role-impersonation'
+import { useRoleImpersonationStateSnapshot } from 'state/role-impersonation-state'
+import { DropdownMenuSeparator, cn } from 'ui'
+import { AnonIcon, AuthenticatedIcon, ServiceRoleIcon } from './Icons'
+import RoleImpersonationRadio from './RoleImpersonationRadio'
+import UserImpersonationSelector from './UserImpersonationSelector'
+
+export interface RoleImpersonationSelectorProps {
+  serviceRoleLabel?: string
+  padded?: boolean
+}
+
+const RoleImpersonationSelector = ({
+  serviceRoleLabel,
+  padded = true,
+}: RoleImpersonationSelectorProps) => {
+  const state = useRoleImpersonationStateSnapshot()
+
+  const [selectedOption, setSelectedOption] = useState<PostgrestRole | undefined>(() => {
+    if (
+      state.role?.type === 'postgrest' &&
+      (state.role.role === 'anon' || state.role.role === 'authenticated')
+    ) {
+      return state.role.role
+    }
+
+    return 'service_role'
+  })
+
+  const isAuthenticatedOptionFullySelected = Boolean(
+    selectedOption === 'authenticated' &&
+      state.role?.type === 'postgrest' &&
+      state.role.role === 'authenticated' &&
+      state.role.user
+  )
+
+  function onSelectedChange(value: PostgrestRole) {
+    if (value === 'service_role') {
+      // do not set a role for service role
+      // as the default role is the "service role"
+      state.setRole(undefined)
+    }
+
+    if (value === 'anon') {
+      state.setRole({
+        type: 'postgrest',
+        role: value,
+      })
+    }
+
+    setSelectedOption(value)
+  }
+
+  return (
+    <>
+      <div className={cn('flex flex-col gap-3', padded ? 'p-5' : 'pb-5')}>
+        <h2 className="text-foreground text-base">Database connection settings</h2>
```

Comment:
im wondering should we use text-sm here and text-xs for the description below? Where this popover is rendered, the text size in this is relatively larger to the other popovers on the pages

e.g sql editor
![image](https://github.com/supabase/supabase/assets/19742402/927395d3-11f8-4ec3-8334-58d14b36c82c)
![image](https://github.com/supabase/supabase/assets/19742402/f5831334-21ce-487a-98b4-978d9eb8f34c)

e.g  table editor
![image](https://github.com/supabase/supabase/assets/19742402/2dac2e4d-6e66-4964-bfb0-3f70c4cdb84b)
![image](https://github.com/supabase/supabase/assets/19742402/32df0976-a3ed-4cf1-abe3-47c68bbad0ce)

Reply:
Do it on another pass.

we need to fix all the font sizes anyway. rather do it as another update,